### PR TITLE
🐛 Attempt to fix the crash on sending heartbeats

### LINKED ISSUE #34: Server crashes with lots of player traffic?
I'm not sure if this is the actual cause of it, but the server seems to just crash when there is a lot of player traffic at one time.

I am attaching the logfile with the max verbosity so you can see what it says.

[MM_server-log_file.txt](https://github.com/RigsOfRods/ror-server/files/614009/MM_server-log_file.txt)

### PR DESCRIPTION
**Pull request by ulteq**
Fixed #34

### CODE REVIEW

File: source/server/http.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ namespace Http {
                         method, host, url, result.get_error().c_str());
             return false;
         }
-        char query[2000] = {0};
+        char query[18000] = {0};
```
Comment (paroj) [inline source/server/http.cpp]: 17MB on the stack? maybe use vector<char>?

Comment (ulteq) [reply source/server/http.cpp]: 18KB or am I blind?

Comment (ulteq) [reply source/server/http.cpp]: The backtrace of the crash looked like this:
```
*** stack smashing detected ***: <unknown> terminated

Thread 1 "rorserver" received signal SIGABRT, Aborted.
0x00007ffff7abbd7f in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  in raise () from /usr/lib/libc.so.6
#1  in abort () from /usr/lib/libc.so.6 
#2  in __libc_message () from /usr/lib/libc.so.6 
#3  in __fortify_fail_abort () from /usr/lib/libc.so.6 
#4  in __stack_chk_fail () from /usr/lib/libc.so.6 
#5  in Http::RequestRaw (method=0x55555558f144 "PUT", host=0x5555555b0e70 "multiplayer.rigsofrods.org", url=0x55555559e8d8 <s_master_server+56> "//server-list", content_type=0x55555558f760 "application/json", 
    payload=0x5555555c5100 "{\n   \"challenge\" : \"\",\n   \"users\" : [\n      {\n         \"client_id\" : 1,\n         \"ip_address\" : \"127.0.0.1\",\n         \"is_admin\" : 0,\n         \"is_bot\" : 0,\n         \"is_mod\" : 0,\n         \"is_ranked\""..., out_response_buffer=0x7fffffffcde0 "HTTP/1.1 200 OK\r\nDate: Wed, 09 Jan 2019 12:50:45 GMT\r\nServer: Apache\r\nContent-Length: 40\r\nContent-Type: application/json; charset: utf-8\r\n\r\n{\"result\":true,\"message\":\"Heartbeat OK\"}", 
    response_buf_len=5000) at /usr/include/c++/8.2.1/bits/basic_string.h:2290
#6  in Http::Request (method=<optimized out>, host=<optimized out>, url=<optimized out>, content_type=content_type@entry=0x55555558f760 "application/json", payload=<optimized out>, response=0x20205f7165746c75, 
    at .../server/http.cpp:100
#7  in MasterServer::Client::HttpRequest (this=this@entry=0xa5d2020200a7d20, method=<optimized out>, payload=<optimized out>, out_response=out_response@entry=0x7fffffffe1d0) at /usr/include/c++/8.2.1/bits/basic_string.h:2290
#8  in MasterServer::Client::SendHeatbeat (this=0xa5d2020200a7d20, this@entry=0x55555559e8a0 <s_master_server>, user_list=...) at /usr/include/c++/8.2.1/bits/basic_string.h:2290
#9  in main (argc=<optimized out>, argv=<optimized out>) at .../server/rorserver.cpp:362
```

Comment (paroj) [reply source/server/http.cpp]: > 18KB or am I blind?

yes you are right. Although with a vector you could resize to the actual/ arbitrary size.

Comment (ulteq) [reply source/server/http.cpp]: -> https://github.com/RigsOfRods/ror-server/pull/82

**Pull request merged**